### ISSUE TITLE
backupwallet fails with multi wallet

### ISSUE THREAD

**Issue by domob1812** [xaya/xaya#48]
The `backupwallet` RPC command seems to not correctly select the wallet when using multi-wallet, at least on Windows.

**Comment by xaya** [comment]
It seems this is happening on linux as well.

It is not correctly selecting the source wallet.dat.

`xaya-cli -rpcwallet=game.dat backupwallet "/home/xxxx/wallet.dat"`

```
error code: -4
error message:
Error: Wallet backup failed!
```
in debug.log

```
2018-07-14T14:24:39Z Received a POST request for /wallet/game.dat from 127.0.0.1:35566
2018-07-14T14:24:39Z ThreadRPCServer method=backupwallet user=__cookie__
2018-07-14T14:24:39Z error copying wallet.dat to /home/xxxx/wallet.dat - boost::filesystem::equivalent: Operation not permitted: "/home/xxxx/.xaya/wallets/wallet.dat", "/home/xxxx/wallet.dat"
```

Identical to windows. 

The source wallet in this case should be 
`/home/xxxx/.xaya/wallets/game.dat/wallet.dat`

If you are testing, make sure you do not have a wallet in the default location (where it is trying to copy from), as it will backup this and look like it's worked.

This could cause people to lose coins as they may believe they have successfully backed up their multiwallets (maybe in Bitcoin too, although I have not tested there).

**Comment by xaya** [comment]
looks like the problem is in bitcoin as well. 

`2018-07-14T14:52:59Z error copying wallet.dat to /home/xxxx/wallet.dat - boost::filesystem::equivalent: Operation not permitted: "/home/xxxx/.bitcoin/wallets/wallet.dat", "/home/xxxx/wallet.dat"`

**Comment by domob1812** [comment]
Thanks for testing - then please open an issue for Bitcoin and post the link here.  Let's see what Bitcoin devs say about it - if they acknowledge the issue but don't immediately have time to fix it, I'm happy to debug and fix it for upstream Bitcoin.

**Comment by domob1812** [comment]
Upstream fix is in https://github.com/bitcoin/bitcoin/pull/13667.

**Comment by domob1812** [comment]
Fixed on `1.0` in 89951e00cf1caf445ccad37ecb5a6a4a0d393c63.  For `master`, we'll fix it through the upstream patch.  I'll keep this issue open until that is merged.

**Comment by domob1812** [comment]
Upstream PR https://github.com/bitcoin/bitcoin/pull/13667 was merged, and that will make its way to Xaya with the next regular upstream merges - so we can close this now.
